stdlib/test/collections/test_list.mojo

```diff
@@ -759,6 +759,32 @@ def test_list_contains():
     # assert_equal(List(0,1) in y,False)
 
 
+def test_list_eq():
+    var a = List[Int](1, 2, 3)
+    var b = List[Int](1, 2, 3)
+    var c = List[Int](0, 1, 2)
+    var d = List[Int](0, 1, 2, 3)
+    assert_true(a == b)
+    assert_false(a == c)
+    assert_false(a == d)
```

Comment:
```suggestion
    assert_false(a == d)
    assert_not_equal(a, d)
```

Reply:
```assert_false``` require the arguments to be ```Testable``` which includes ```EqualityComparable```,

this would not work because List is not ```EqualityComparable``` and ```Stringable```;

What the ```__eq__``` implementation does is to exist if T of a specific list is ```ComparableCollectionElement```.

&nbsp;

(see ```fn __str__[U: RepresentableCollectionElement](self: List[U]) -> String:``` of ```List```)





It would require the "conditional conformance"

Reply:
Thanks for the info! I'm assuming that at some stage `assert_equal` will work for collections? I wonder If it's worth (in a separate MR) attempting to enhance the testing library to work for `collections` that contain `ComparableCollectionElement`?

Reply:
Not sure, @ChristopherLR , if you think it's a good idea you can always develop it